Adding queue name and change the database schema

thoven87 opened the pull request:
* Adding queue name
* Migrate the Postgres driver to `swift_jobs` schema
* Drop all previous migrations 

Renamed the following tables:

* _hb_pg_job_queue -> swift_jobs.queues
* _hb_pg_jobs -> swift_jobs.jobs
* _hb_pg_job_queue_metadata -> swift_jobs.queues_metadata

Renamed the following columns:

* `createdAt` -> `created_at`
* `lastModified` -> `last_modified`

Added new indexes which reflects the current schema.

adam-fowler commented on Sources/JobsPostgres/Migrations/CreateSwiftJobsMigrations.swift:
```diff
@@ -0,0 +1,88 @@
+
+        try await connection.query(
+            """
+            CREATE TABLE IF NOT EXISTS swift_jobs.jobs (
```
Should swift_jobs.jobs also include the queue name in it?

Currently the queue initialisation can re-queue jobs based off their status. It does it for all jobs, not just the ones in that queue. It should really only do it for the jobs in that queue. To do this would mean we need to keep a record of which queue a job is in.

thoven87 replied:
Good point, I added a new column which does just that.

adam-fowler reviewed (commented):
In general looks good. Not sure we should keep breaking changes details in README. There are a number of breaking changes we have introduced and we should detail those in the release notes.

thoven87 commented:
> In general looks good. Not sure we should keep breaking changes details in README. There are a number of breaking changes we have introduced and we should detail those in the release notes.

Noted, I removed the breaking changed notes. I will tag the notes in a breaking changed issue.

adam-fowler commented on Sources/JobsPostgres/PostgresJobsQueue.swift:
```diff
@@ -362,26 +362,26 @@ public final class PostgresJobQueue: JobQueueDriver {
+    func getJobs(withStatus status: Status) async throws -> [(id: JobID, queue: String)] {
         let stream = try await self.client.query(
-            "SELECT id FROM _hb_pg_jobs WHERE status = \(status) FOR UPDATE SKIP LOCKED",
+            "SELECT id, queue_name FROM swift_jobs.jobs WHERE status = \(status)",
```
Need to check for queue name here. We don't want to be processing jobs from other queues
```suggestion
            "SELECT id FROM swift_jobs.jobs WHERE status = \(status) AND queue_name = \(configuration.queueName)",
```
You'll need to fix up other related code.

adam-fowler commented on Sources/JobsPostgres/PostgresJobsQueue.swift:
```diff
@@ -362,26 +373,42 @@ public final class PostgresJobQueue: JobQueueDriver {
+                queue_name
+            FROM swift_jobs.jobs
+            WHERE status = \(status) AND queue_name = \(configuration.queueName)
+            """,
```
We don't need to get the queue name here. It is always `configuration.queueName`

adam-fowler reviewed (commented):
One minor thing, but otherwise we are good

thoven87 commented:
> One minor thing, but otherwise we are good

I addressed the feedback on the most recent commit.
